src/registry.rs

```diff
@@ -346,6 +346,19 @@ pub trait NpmRegistryApi {
     name: &str,
   ) -> Result<Arc<NpmPackageInfo>, NpmRegistryPackageInfoLoadError>;
 
+  /// Optional method an implementer can use to start downloading a package
+  /// name and version and doing a basic setup of the node_modules directory.
+  ///
+  /// deno_npm will call this method indistriminately for every package nv it sees.
+  /// It is up to the implementer to not do extra work.
+  fn preload_package_nv(
+    &self,
+    _nv: &PackageNv,
+    _dist: &NpmPackageVersionDistInfo,
```

Comment:
Actually, this needs to provide the full version information so that the implementer can decide if the package is relevant for the current system.